resource_quota.go

```diff
@@ -0,0 +1,218 @@
+		[]string{
+			"name",
+			"namespace",
+			"resource",
+			"type",
+		}, nil,
+	)
+	descResourceQuotaStorage = prometheus.NewDesc(
+		"kube_resource_quota_storage",
+		"Information about storage resource quota.",
+		[]string{
+			"name",
+			"namespace",
+			"resource",
+			"type",
+		}, nil,
+	)
+
+	resources = map[v1.ResourceName]*prometheus.Desc{
+		v1.ResourceCPU:                    descResourceQuotaCPU,
+		v1.ResourceMemory:                 descResourceQuotaMemory,
+		v1.ResourceStorage:                descResourceQuotaStorage,
+		v1.ResourcePods:                   descResourceQuotaPods,
+		v1.ResourceServices:               descResourceQuotaServices,
+		v1.ResourceReplicationControllers: descResourceQuotaReplicationControllers,
+		v1.ResourceQuotas:                 descResourceQuotaResourceQuotas,
+		v1.ResourceSecrets:                descResourceQuotaSecrets,
+		v1.ResourceConfigMaps:             descResourceQuotaConfigMaps,
+		v1.ResourcePersistentVolumeClaims: descResourceQuotaPersistentVolumeClaims,
+		v1.ResourceServicesNodePorts:      descResourceQuotaNodePorts,
+		v1.ResourceServicesLoadBalancers:  descResourceQuotaLoadBalancers,
+	}
+)
+
+type resourceQuotaStore interface {
+	List() (v1.ResourceQuotaList, error)
+}
+
+// resourceQuotaCollector collects metrics about all resource quotas in the cluster.
+type resourceQuotaCollector struct {
+	store resourceQuotaStore
+}
+
+// Describe implements the prometheus.Collector interface.
+func (rqc *resourceQuotaCollector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- descResourceQuotaCPU
+	ch <- descResourceQuotaMemory
+	ch <- descResourceQuotaPods
+	ch <- descResourceQuotaServices
+	ch <- descResourceQuotaReplicationControllers
+	ch <- descResourceQuotaResourceQuotas
+	ch <- descResourceQuotaSecrets
+	ch <- descResourceQuotaConfigMaps
+	ch <- descResourceQuotaPersistentVolumeClaims
+	ch <- descResourceQuotaNodePorts
+	ch <- descResourceQuotaLoadBalancers
+	ch <- descResourceQuotaStorage
+}
+
+// Collect implements the prometheus.Collector interface.
+func (rqc *resourceQuotaCollector) Collect(ch chan<- prometheus.Metric) {
+	resourceQuota, err := rqc.store.List()
+	if err != nil {
+		glog.Errorf("listing resource quotas failed: %s", err)
+		return
+	}
+
+	for _, rq := range resourceQuota.Items {
+		rqc.collectResourceQuota(ch, rq)
+	}
+}
+
+func (rqc *resourceQuotaCollector) collectResourceQuota(ch chan<- prometheus.Metric, rq v1.ResourceQuota) {
+	addGauge := func(desc *prometheus.Desc, v float64, lv ...string) {
+		lv = append([]string{rq.Name, rq.Namespace}, lv...)
+		ch <- prometheus.MustNewConstMetric(desc, prometheus.GaugeValue, v, lv...)
+	}
+
+	// Add capacity and allocatable resources if they are set.
+	addResource := func(d *prometheus.Desc, res v1.ResourceList, n v1.ResourceName, labels ...string) {
+		if v, ok := res[n]; ok {
+			addGauge(d, float64(v.MilliValue())/1000, labels...)
+		}
+	}
+	for res, desc := range resources {
```

Comment:
I'm not sure to understand correctly. 
Do you mean we should only have a `kube_resource_quota` metric and using the `resource` label ?
Something like : 
```
kube_resource_quota{name="quotaTest",namespace="testNS",resource="cpu",type="hard"} 4.3
kube_resource_quota{name="quotaTest",namespace="testNS",resource="cpu",type="used"} 2.1
kube_resource_quota{name="quotaTest",namespace="testNS",resource="memory",type="hard"} 2.1e+09
kube_resource_quota{name="quotaTest",namespace="testNS",resource="memory",type="used"} 5e+08
```